xdsl/transforms/riscv_register_allocation.py

```diff
@@ -1,9 +1,80 @@
 from abc import ABC
+from collections import OrderedDict
 from dataclasses import dataclass
+from typing import Any, Sequence
 from xdsl.dialects.builtin import ModuleOp
 from xdsl.dialects.riscv import Register, RegisterType, RISCVOp
-from xdsl.ir import MLContext
+from xdsl.ir import MLContext, SSAValue
 from xdsl.passes import ModulePass
+from xdsl.transforms.experimental.live_range import LiveRange
+
+
+class RegisterSet:
+    # The complete set of registers
+    registers: Sequence[str]
+    # Registers that are free to be allocated
```

Comment:
nit: this would be useful as documentation comments so you see the doc when you hover over the usage